src/test/java/edu/ksu/canvas/attendance/submitter/AssignmentValidatorUTest.java

```diff
@@ -0,0 +1,148 @@
+    private static final Long COURSE_ID = 2121212121L;
+    private static final String OAUTH_STRING = "sdfsdfSDFSDFsdfsdFSDFsdfSDFSDgfsdSFDFSDF";
+    private static final long ASSIGNMENT_ID = 8484848484L;
+    private static final String ASSIGNMENT_NAME = "NAME OF THE ASSIGNMENT";
+    private static final Double ASSIGNMENT_POINTS = 100.0;
+    private static final long CANVAS_ASSIGNMENT_ID = 514514514L;
+    private static final String SECTION_NAME = "SECTION NAME";
+    private static final long CANVAS_COURSE_ID = 11111L;
+    private static final long SECTION_ID = 121212L;
+    private static final long CANVAS_SECTION_ID = 212121L;
+
+    private AssignmentValidator assignmentValidator;
+
+    @Mock
+    private CanvasApiWrapperService canvasApiWrapperService;
+
+    private NonRefreshableOauthToken oauthToken;
+    private AttendanceAssignment attendanceAssignment;
+    private Optional<Assignment> assignmentOptional;
+    private Assignment assignment;
+    private AttendanceAssignment assignmentConfigurationFromSetup;
+    private AttendanceSection attendanceSection;
+
+    @Before
+    public void setup() {
+
+        assignmentValidator = new AssignmentValidator();
+
+        oauthToken = new NonRefreshableOauthToken(OAUTH_STRING);
+
+        assignmentConfigurationFromSetup = new AttendanceAssignment();
+        assignmentConfigurationFromSetup.setAbsentPoints(0.0);
+        assignmentConfigurationFromSetup.setAssignmentPoints(ASSIGNMENT_POINTS);
+        assignmentConfigurationFromSetup.setExcusedPoints(0.0);
+        assignmentConfigurationFromSetup.setPresentPoints(100.0);
+        assignmentConfigurationFromSetup.setTardyPoints(0.0);
+
+        attendanceSection = new AttendanceSection();
+        attendanceSection.setName(SECTION_NAME);
+        attendanceSection.setCanvasCourseId(CANVAS_COURSE_ID);
+        attendanceSection.setSectionId(SECTION_ID);
+        attendanceSection.setCanvasSectionId(CANVAS_SECTION_ID);
+
+        attendanceAssignment = new AttendanceAssignment();
+        attendanceAssignment.setAssignmentId(ASSIGNMENT_ID);
+        attendanceAssignment.setAssignmentName(ASSIGNMENT_NAME);
+        attendanceAssignment.setCanvasAssignmentId(CANVAS_ASSIGNMENT_ID);
+        attendanceAssignment.setGradingOn(true);
+        attendanceAssignment.setAssignmentPoints(ASSIGNMENT_POINTS);
+        attendanceAssignment.setAbsentPoints(0.0);
+        attendanceAssignment.setExcusedPoints(0.0);
+        attendanceAssignment.setTardyPoints(0.0);
+        attendanceAssignment.setPresentPoints(100.0);
+        attendanceAssignment.setAttendanceSection(attendanceSection);
+        attendanceAssignment.setStatus(AttendanceAssignment.Status.UNKNOWN);
+
+
+        assignment = new Assignment();
+        assignment.setPointsPossible(ASSIGNMENT_POINTS);
+        assignmentOptional = Optional.of(assignment);
+
+    }
+
+    @Test
+    public void validateAttendanceAssignmentHappyPath() throws IOException, AttendanceAssignmentException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this method to throw at most one checked exception instead of: java.io.IOException, edu.ksu.canvas.attendance.exception.AttendanceAssignmentException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1160)